### PR TITLE
Bind interactive debugger to socket

### PR DESCRIPTION
**Pull request by alexcb**
Rather than using tcp to run the interactive debugger, use a
socket-based connection. This is more secure as it doesn't expose any
ports on the network (and also prevents firewalls from disallowing
connections).

### CODE REVIEW

File: debugger/server/debugger_server.go [hunk 1]
```diff
@@ -107,16 +109,17 @@ func (ds *DebugServer) Start() (string, error) {
 			}
 		}
 	}()
-	return l.Addr().String(), nil
+	return nil
 }
 
 // Stop stops the server
 func (ds *DebugServer) Stop() {
+	fmt.Printf("stopping debug server")
```
Comment (vladaionescu) [inline debugger/server/debugger_server.go:117]: We should use `conslogging` and a prefix (eg `WithPrefix("debugger")`) for messages like this, so that it looks a little cleaner.

... though I wonder if the user even needs to know about the debug server. Maybe we don't say anything unless there's an error?

File: earthfile2llb/converter.go [hunk 2]
```diff
@@ -672,11 +673,21 @@ func (c *Converter) internalRun(ctx context.Context, args []string, secretKeyVal
 		}
 	}
 
+	finalOpts = append(finalOpts,
+		llb.AddMount("/usr/bin/earth_debugger",
+			llb.Image(c.debuggerImage, llb.MarkImageInternal, llb.ResolveModePreferLocal, llb.Platform(llbutil.TargetPlatform)),
```
Comment (vladaionescu) [inline earthfile2llb/converter.go:678]: I have an idea... if we host-bind the debugger binary too, then we wouldn't bust the cache if the debugger binary changes. What do you think? (This can be a separate PR though)

File: buildkitd/buildkitd.go [hunk 3]
```diff
@@ -192,10 +192,12 @@ func Start(ctx context.Context, image string, settings Settings, reset bool) err
 	}
 	env := os.Environ()
 	cacheMount := fmt.Sprintf("%s:/tmp/earthly:delegated", settings.TempDir)
+	runMount := fmt.Sprintf("%s:/run/earthly:delegated", settings.RunDir)
```
[the remark this answers is not in the document]
Comment (alexcb) [reply buildkitd/buildkitd.go:195]: good idea, i'll switch it to `consistent`; we'll need to do a bit more macOS testing for this.

File: debugger/server/debugger_server.go (hunk 1 above)
Comment (alexcb) [reply debugger/server/debugger_server.go:117]: whoops, this was some left-over debugging code that slipped in. I'll remove it.

File: earthfile2llb/converter.go (hunk 2 above)
Comment (alexcb) [reply earthfile2llb/converter.go:678]: that would also make distributing the debugger binary easier. Currently we have to manually push it to docker hub, this would greatly improve that process.

**Comment by alexcb** [comment]
Good find on the macOS socket issue. It would be really hacky, but maybe we could abuse the ssh agent socket and look for a special non-ssh handshake which would then forward data to the debugger shell, otherwise we would forward data to the regular ssh agent.  This is a last-resort sort of idea though.
